Mathlib/RingTheory/MvPolynomial/MonomialOrder.lean

```diff
@@ -331,17 +365,54 @@ theorem degree_smul {r : R} (hr : IsRegular r) {f : MvPolynomial σ R} :
   rw [← zero_add (degree m f), ← degree_C r, coeff_mul_of_degree_add]
   simp [lCoeff, hr.left.mul_left_eq_zero_iff, hf]
 
-theorem eq_C_of_degree_eq_zero {f : MvPolynomial σ R} (hf : m.degree f = 0) :
-    f = C (m.lCoeff f) := by
-  ext d
-  simp only [lCoeff, hf]
+theorem degree_prod_le {ι : Type*} {P : ι → MvPolynomial σ R} {s : Finset ι} :
+    m.degree (s.prod P) ≼[m] s.sum (fun i ↦ m.degree (P i)) := by
   classical
-  by_cases hd : d = 0
-  · simp [hd]
-  · rw [coeff_C, if_neg (Ne.symm hd)]
-    apply coeff_eq_zero_of_lt (m := m)
-    rw [hf, map_zero, lt_iff_le_and_ne, ne_eq, eq_comm, EmbeddingLike.map_eq_zero_iff]
-    exact ⟨bot_le, hd⟩
+  induction s using Finset.induction_on with
+  | empty =>
+    simp only [Finset.prod_empty, Finset.sum_empty]
+    rw [← C_1, m.degree_C, map_zero]
+  | @insert a s has hrec =>
+    rw [Finset.prod_insert has, Finset.sum_insert has]
+    apply le_trans degree_mul_le
+    simp only [map_add, add_le_add_iff_left, hrec]
+
+theorem coeff_prod_of_sum_degree {ι : Type*} (P : ι → MvPolynomial σ R) (s : Finset ι) :
+    coeff (s.sum fun i ↦ m.degree (P i)) (s.prod P) = s.prod fun i ↦ m.lCoeff (P i) := by
+  classical
+  induction s using Finset.induction_on with
+  | empty => simp
+  | @insert a s has hrec =>
+    simp only [Finset.prod_insert has, Finset.sum_insert has]
+    rw [coeff_mul_of_add_of_degree_le (le_of_eq rfl) degree_prod_le]
+    exact congr_arg₂ _ rfl hrec
+
+-- TODO : it suffices that all leading coefficients but one are regular
+theorem degree_prod_of_regular {ι : Type*}
+    {P : ι → MvPolynomial σ R} {s : Finset ι} (H : ∀ i ∈ s, IsRegular (m.lCoeff (P i))) :
+    m.degree (s.prod P) = s.sum (fun i ↦ m.degree (P i)) := by
+  by_cases hR : Nontrivial R
+  · apply m.toSyn.injective
+    · refine le_antisymm degree_prod_le (m.le_degree ?_)
+      rw [mem_support_iff, m.coeff_prod_of_sum_degree]
+      exact (IsRegular.prod H).ne_zero
+  · -- case : Trivial R
+    simp [degree_of_not_nontrivial hR]
+
+theorem degree_prod [IsDomain R] {ι : Type*} {P : ι → MvPolynomial σ R} {s : Finset ι}
+    (H : ∀ i ∈ s, P i ≠ 0) :
+    m.degree (s.prod P) = s.sum (fun i ↦ m.degree (P i)) := by
```

Comment:
Can you use the `∑` and `∏` notations, here and everywhere else?

Reply:
done